src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionTextCache.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
```

Comment:
This should be in the same namespace as `ExpressionHelper`. I'm not positive why that one isn't `.Internal` but it isn't.